mountpoint-s3-client/src/s3_crt_client/get_object.rs

```diff
@@ -114,11 +116,14 @@ impl S3CrtClient {
                 let _ = sender.unbounded_send(Ok((offset, data.into())));
             },
             move |result| {
-                // FIXME - Ideally we'd include a reason why we failed here.
-                object_headers_setter_on_finish.or_set(Err(ObjectHeadersError::UnknownError));
                 if result.is_err() {
-                    Err(parse_get_object_error(result).map(ObjectClientError::ServiceError))
+                    let err = parse_get_object_error(result);
```

Comment:
This is problematic: the result has not been processed yet - see [these](https://github.com/awslabs/mountpoint-s3/blob/b61f4b966f616ed3d231857403946149520aad2a/mountpoint-s3-client/src/s3_crt_client.rs#L514) [comments](https://github.com/awslabs/mountpoint-s3/blob/b61f4b966f616ed3d231857403946149520aad2a/mountpoint-s3-client/src/s3_crt_client.rs#L648).

I think this is the same mistake that causes issues for PUT: https://github.com/awslabs/mountpoint-s3/issues/1007.

Unless we review the callbacks, the correct thing to do would be to wait for the request to finish and return the fully parsed error.